Firestore/core/src/firebase/firestore/immutable/llrb_node.h

```diff
@@ -258,6 +288,63 @@ LlrbNode<K, V> LlrbNode<K, V>::InnerInsert(const K& key,
   return result;
 }
 
+template <typename K, typename V>
+template <typename Comparator>
+LlrbNode<K, V> LlrbNode<K, V>::erase(const K& key,
+                                     const Comparator& comparator) const {
+  LlrbNode root = InnerErase(key, comparator);
+  root.FixRootColor();
+  return root;
+}
+
+template <typename K, typename V>
+template <typename Comparator>
+LlrbNode<K, V> LlrbNode<K, V>::InnerErase(const K& key,
+                                          const Comparator& comparator) const {
+  if (empty()) {
+    // Empty node already frozen
+    return LlrbNode{};
+  }
+
+  LlrbNode n = Clone();
+
+  bool descending = comparator(key, n.key());
+  if (descending) {
+    if (!n.left().empty() && !n.left().red() && !n.left().left().red()) {
```

Comment:
Optional: perhaps
```cpp
if (!n.left().empty() && n.left().black() && n.left().left().black()) {
```
would be a little clearer (that is, `black()` vs `!red()`)?